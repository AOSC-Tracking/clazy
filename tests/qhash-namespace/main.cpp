#include <QtCore/QString>

struct A { };
uint qHash(A) { return 0; }; // OK

namespace NS {
    typedef int IntFoo;
    struct B { struct B2 {}; struct B3 {}; };
    uint qHash(B) { return 0; }; // OK
    uint qHash(B::B2) { return 0; }; // OK
    namespace NS2 {
        struct C {};
        uint qHash(C) { return 0; }; // OK
    }
    uint qHash(NS2::C) { return 0; }; // Warn
    uint qHash(::A) { return 0; } // Warn
    uint qHash(B::B3) { return 0; }; // OK
}
uint qHash(NS::B) { return 0; } // Warn
uint qHash(NS::B *) { return 0; } // Warn
uint qHash(NS::B::B3) { return 0; }; // Warn
uint qHash(NS::IntFoo) { return 0; }
