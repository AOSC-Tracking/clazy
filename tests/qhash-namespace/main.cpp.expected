qhash-namespace/main.cpp:15:5: warning: Move qHash(NS::NS2::C) to NS::NS2 namespace for ADL lookup [-Wclazy-qhash-namespace]
qhash-namespace/main.cpp:16:5: warning: Move qHash(A) out of namespace NS [-Wclazy-qhash-namespace]
qhash-namespace/main.cpp:19:1: warning: Move qHash(NS::B) into NS namespace for ADL lookup [-Wclazy-qhash-namespace]
qhash-namespace/main.cpp:20:1: warning: Move qHash(NS::B *) into NS namespace for ADL lookup [-Wclazy-qhash-namespace]
qhash-namespace/main.cpp:21:1: warning: Move qHash(NS::B::B3) into NS namespace for ADL lookup [-Wclazy-qhash-namespace]
qhash-namespace/main.cpp:22:1: warning: Move qHash(NS::IntFoo) into NS namespace for ADL lookup [-Wclazy-qhash-namespace]
